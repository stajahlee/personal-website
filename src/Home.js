import withRoot from './modules/withRoot';
// --- Post bootstrap -----
import React from 'react';
import Categories from './modules/views/Categories';
import GetInTouch from './modules/views/GetInTouch';
import AppFooter from './modules/views/AppFooter';
import Intro from './modules/views/Intro';
import FrontAppBar from './modules/views/FrontAppBar';

function Index() {
  return (
    <React.Fragment>
      <FrontAppBar />
      <Intro />
      <Categories />
      <GetInTouch />
      <AppFooter />
    </React.Fragment>
  );
}

export default withRoot(Index);
