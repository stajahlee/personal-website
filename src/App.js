import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from "react-router-dom";
import Home from './Home';
import Paintings from './modules/views/Paintings'
import './App.css';

function App() {
  return (
    <Router basename={process.env.PUBLIC_URL}>
      <div>
        {/* A <Switch> looks through its children <Route>s and
            renders the first one that matches the current URL. */}
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/home" component={Home} />
          <Route path="/paintings" component={Paintings} />
          <Route path={new RegExp('/.*')} component={Home} />
        </Switch>
      </div>
    </Router>
  );
}

export default App;
