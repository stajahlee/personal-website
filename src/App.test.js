import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import Home from './Home';

describe('AppComponent', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
   });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Home component', () => {
    const home = <Home />;
    expect(wrapper.contains(home)).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });
});
