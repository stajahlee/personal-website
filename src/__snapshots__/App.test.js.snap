// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent renders Home component 1`] = `ShallowWrapper {}`;

exports[`AppComponent should render correctly 1`] = `ShallowWrapper {}`;
